### PR TITLE
Fix issue with service is not properly cleaned up on Alpine Linux images

### PR DESCRIPTION
**Pull request by nghiant2710**
Connects to https://github.com/resin-io/hq/issues/987.
After checking user's log and Dockerfile, this issue happens when container starts and can only be reproduced by creating a process and echoing its PID to `/var/run/resinapp.pid` before starting OpenRC. 

When OpenRC starts resin service, it will  first check whether the process with PID in `/var/run/resinapp.pid` is running or not (if that file exists) and the error will be thrown in case it's running. It seems weird that the issue can happen right after the container starts so I think the possible fix here is to make sure that the pidfile and process don't exist.

### CODE REVIEW

**Comment by petrosagg** [comment]
This doesn't sound like a problem. Only the first process should get the signal

**Comment by nghiant2710** [comment]
@petrosagg, the current implementation `of resinApp.sh (`#!/bin/bash\n exec $@`) will just replace the running shell (resinApp.sh) with whatever users put in the CMD (first process), however, OpenRC will only kill `resinApp.sh` on service restart or stop since it's in the service configuration file and so the first process will never receive the signal. This PR will fix the issue by trapping resinApp exit and send signal to the first process.

**Comment by petrosagg** [comment]
@nghiant2710 I don't understand the problem. After `exec` `resinApp.sh` doesn't exist. OpenRC will send the signal to the exec'd process

**Comment by nghiant2710** [comment]
@petrosagg OpenRC will only send the signal to the process with pid in pidfile (/var/run/resinapp.pid) which is resinApp.sh (https://github.com/resin-io-library/base-images/blob/master/alpine/resin#L7-L16). If it doesn't exit then it won't send the signal to the exec'd process.

**Comment by petrosagg** [comment]
When you start `resinApp.sh`, it gets a PID. When it does `exec foobar`, then its process gets replaced with `foobar`, but foobar will have **the same** PID. If you observe that the PID is changing is because whatever you're exec'ing is forking a child and exiting from the partent

**Comment by nghiant2710** [comment]
@petrosagg you're right. I've checked and updated the PR (please check my first comment)

**Comment by imrehg** [comment]
@petrosagg @nghiant2710 ping on this issue ^^

File: alpine/aarch64/3.5/entry.sh [hunk 1]
```diff
@@ -44,6 +44,12 @@ function init_openrc()
 	printf '%q ' "$@" >> /etc/resinApp.sh
 	chmod +x /etc/resinApp.sh
 
+	# Make sure there is no pidfile or process running before starting.
+	if [ -f "/var/run/resinapp.pid" ]; then
+		kill $(cat /var/run/resinapp.pid) &> /dev/null
```
Comment (petrosagg) [inline alpine/aarch64/3.5/entry.sh]: You don't have to try and kill this. The problem was most likely caused by a container restart where the pid file persisted in the writable layer. So you can just leave the rm here

Comment (nghiant2710) [reply alpine/aarch64/3.5/entry.sh]: Updated!

**Comment by pimterry** [comment]
@petrosagg @nghiant2710 ping on this issue, we've got [a support thread](https://app.frontapp.com/open/cnv_6s7pxj) waiting on updates on it, looks like it's waiting for rereview.

**Pull request merged**